src/indexer/apy-snapshots-indexer.ts

```diff
@@ -0,0 +1,221 @@
+import { isNil } from '@tsed/core';
+import { TransactWriteItem } from 'aws-sdk/clients/dynamodb';
+import flatten from 'lodash/flatten';
+import { transactWrite } from '../aws/dynamodb-utils';
+import { loadChains } from '../chains/chain';
+import { bscSetts } from '../chains/config/bsc.config';
+import { Chain } from '../chains/config/chain.config';
+import { ethSetts } from '../chains/config/eth.config';
+import { ChainNetwork } from '../chains/enums/chain-network.enum';
+import { APY_SNAPSHOTS_DATA, PANCAKESWAP_URL, Protocol, SUSHISWAP_URL } from '../config/constants';
+import { getSwapValueSource } from '../protocols/common/performance-utils';
+import { CachedValueSource } from '../protocols/interfaces/cached-value-source.interface';
+import { ValueSource } from '../protocols/interfaces/value-source.interface';
+import { PancakeSwapService } from '../protocols/pancake/PancakeSwapService';
+import { ProtocolsService } from '../protocols/ProtocolsService';
+import { SushiswapService } from '../protocols/sushi/SushiswapService';
+import { SettDefinition } from '../setts/interfaces/sett-definition.interface';
+
+const BATCH_SIZE = 10;
+
+export async function refreshApySnapshots() {
+  loadChains();
+
+  const rawValueSources = await Promise.all([
+    ...bscSetts.map(async (settDefinition) => {
+      const getBscValueSource = getSettDefinitionValueSource(ChainNetwork.BinanceSmartChain);
```

Comment:
can we wrap these in the same error handling that was added to the sett snapshots so that we can ensure all complete / update if they are successful.

Reply:
not a blocker for now i think. this can be a follow up.